### DISCUSSION TITLE
Temperature-Dependent Core Leakage Power

### DISCUSSION THREAD

**Discussion by anujpathania** [marg-tools/CoMeT#12] [Q&A] [answered]
Hi, 

Below is the screen-grab from the CoMeT paper.

<img width="428" alt="image" src="https://user-images.githubusercontent.com/15923199/219664844-6758fd91-01ef-4e36-b7c0-9352a337265e.png">

Where is the code corresponding to the statement " we use McPAT to extend HotSpot to account for temperature-dependent leakage
power for cores."?

Which files (or commits) should we look at?

**Comment by siddhulokesh** [comment, marked as answer]
Hi Anuj,

Thanks for your question. Please see the file: https://github.com/marg-tools/CoMeT/blob/main/hotspot_tool/temperature.c

The function _calc_core_leakage_ calculates core leakage power consumption. 

Background: Presently, McPAT provides leakage power consumption for 10-degree changes in temperature (such as 60, 70, 80 degrees Celsius). By obtaining an equation (using curve-fitting), we are able to calculate the effect on leakage power even for changes in temperature that are smaller than 10 degrees Celsius.
